### PR TITLE
Media Palette

### PR DESCRIPTION
**Pull request by alinaboghiu**
## What does this change?

Fixes: https://github.com/guardian/dotcom-rendering/issues/5149

Introduces the concept of a "MediaPalette" for use by the fixed/video container - this may eventually become a tag in the Fronts tool same as other palettes but for now it is hardcoded for this container type

## Why?

The video container needs to have a black background, this tends to be the design for audio/video items across the Guardian and is required for the video container. 

## Some Notes

The yellow play button is the latest design requirement for media cards
We may make further tweaks to this in line with any suggestions from @HarryFischer

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |

[before]: https://github.com/guardian/dotcom-rendering/assets/1229808/01c37194-1583-4aef-8de9-ce347aa7dcfe
[after]: https://github.com/guardian/dotcom-rendering/assets/1229808/1764489d-de5b-4a0b-a654-46fc2c83a34e

### CODE REVIEW

File: dotcom-rendering/src/components/Palettes.stories.tsx [hunk 1]
```diff
@@ -167,3 +168,56 @@ export const SpecialReportAltPalette = () => (
...
+	// 	<Carousel
+	// 		heading={"Media Palette"}
+	// 		trails={trails}
+	// 		onwardsSource={'unknown-source'}
+	// 		titleColour={brandAlt[400]}
+	// 		titleHighlightColour={brandAlt[400]}
+	// 		activeDotColour={brandAlt[400]}
+	// 		leftColSize={'compact'}
+	// 	/>
+	// </Section>
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Palettes.stories.tsx:222]: Cleanup?

File: dotcom-rendering/src/lib/decideContainerOverrides.ts [hunk 2]
```diff
@@ -43,44 +46,49 @@ const textCardHeadline = (
...
 
-const textContainerFooter = (
-	containerPalette: Extract<DCRContainerPalette, 'Branded'>,
-): string => {
+const textContainerFooter = (containerPalette: DCRContainerPalette): string => {
 	switch (containerPalette) {
-		case 'Branded':
+		case 'BrandedPalette':
+			return neutral[46];
+		default:
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/decideContainerOverrides.ts:53]: We would usually keep a fall-through behaviour, here… but currently it's not even clear why this isn't simply a constant.

File: dotcom-rendering/src/lib/verticalDivider.ts [hunk 3]
```diff
@@ -22,9 +23,10 @@ export const verticalDivider = (
 				width: 1px;
 				height: 100%;
 				border-left: 1px solid
-					${containerOverrides
-						? containerOverrides.border?.container
-						: border.secondary};
+					${verticalDividerColour ??
+					(containerOverrides
+						? containerOverrides.border.container
+						: border.secondary)};
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/verticalDivider.ts:29]: So you can override the override?

File: dotcom-rendering/index.d.ts [hunk 4]
```diff
@@ -323,8 +323,6 @@ type SmallHeadlineSize =
 
 type MediaType = 'Video' | 'Audio' | 'Gallery';
 
-type LineEffectType = 'labs' | 'dotted' | 'straight';
-
```
Comment (mxdvl) [inline dotcom-rendering/index.d.ts:327]: Is this related?

**Review by mxdvl — COMMENTED** [review]
Would it be possible to extract the refactors—Branded rename, mainly—out of this PR, they make it hard to review the actual change that's happening? The commit history isn't particularly telling either to separate individual changes.

I'm not sure we should allow overriding an override, as we do for the vertical divider. Can you elaborate on why that's needed?

File: dotcom-rendering/src/lib/decideContainerOverrides.ts [hunk 5]
```diff
@@ -43,44 +46,49 @@ const textCardHeadline = (
...
+		case 'BrandedPalette':
 			return labs[400];
+		default:
+			return neutral[46];
 	}
 };
 
 const backgroundContainerLeftColumn = (
-	containerPalette: Extract<DCRContainerPalette, 'Branded'>,
+	containerPalette: Extract<DCRContainerPalette, 'BrandedPalette'>,
```
Comment (jamesgorrie) [inline dotcom-rendering/src/lib/decideContainerOverrides.ts:70]: I'm not 100% sure why these `Extract` type helpers are used, and they seem to be used differently in different situations. Maybe worth looking at in another PR to not distract from the point of this PR.

File: dotcom-rendering/src/components/Carousel.importable.tsx [hunk 6]
```diff
@@ -30,8 +30,13 @@ type Props = {
...
 	onwardsSource: OnwardsSource;
 	leftColSize: LeftColSize;
-	activeDotColour: string;
+	activeDotColour?: string;
+	titleColour?: string;
 	titleHighlightColour: string;
+	verticalDividerColour?: string;
+	arrowColour?: string;
+	arrowBackgroundColour?: string;
+	arrowBackgroundHoverColour?: string;
```
Comment (OllysCoding) [inline dotcom-rendering/src/components/Carousel.importable.tsx:39]: Rather than taking a whole load of colours as props, I think the pattern in DCR would be to pass an optional props of the container palette (`containerPalette?: DCRContainerPalette`), then have the component call decidePalette or `decideContainerOverrides` itself - this is supposed to help us most easily keep track of where we're using colours around the project.

File: dotcom-rendering/src/components/Palettes.stories.tsx (hunk 1 above)
Comment (alinaboghiu) [reply dotcom-rendering/src/components/Palettes.stories.tsx:222]: yes thanks!

**Review by jamesgorrie — COMMENTED** [review]
Dropped a couple of (maybe distracting) thoughts on `MediaType`, but you're pairing with @mxdvl on this, so will leave them there for thought.

File: dotcom-rendering/src/lib/verticalDivider.ts (hunk 3 above)
Comment (mxdvl) [reply dotcom-rendering/src/lib/verticalDivider.ts:29]: Probably worth pulling up one level and only passing the borderColour as to the `verticalDivider` method.

```ts
// capture the logic in one fell swoop and simplify the signatures of leaf methods
const borderColour =
	verticalDividerColour ??
	(containerPalette &&
		decideContainerOverrides(containerPalette).border.container &&
		palette.neutral[86]);
```

File: dotcom-rendering/src/lib/decideContainerOverrides.ts (hunk 5 above)
Comment (mxdvl) [reply dotcom-rendering/src/lib/decideContainerOverrides.ts:70]: Discussed with @alinaboghiu and a potential refactor could use `as const` assertions in a follow-up PR:

```ts
const textCardHeadlineObj = {
	LongRunningPalette: neutral[100],
	LongRunningAltPalette: neutral[7],
	SombrePalette: neutral[100],
	SombreAltPalette: neutral[100],
	InvestigationPalette: neutral[100],
	BreakingPalette: neutral[100],
	EventPalette: brand[300],
	EventAltPalette: brand[300],
	SpecialReportAltPalette: specialReportAlt[100],
	BrandedPalette: undefined,
	MediaPalette: undefined,
} as const satisfies Record<DCRContainerPalette, string | undefined>;
```

File: dotcom-rendering/src/lib/decideContainerOverrides.ts [hunk 7]
```diff
@@ -43,21 +43,23 @@ const textCardHeadline = (
 const textCardStandfirst = textCardHeadline;
 const textCardFooter = textCardHeadline;
 
-const textContainerFooter = (
-	containerPalette: Extract<DCRContainerPalette, 'Branded'>,
-): string => {
+const textContainerFooter = (containerPalette: DCRContainerPalette): string => {
 	switch (containerPalette) {
 		case 'Branded':
 			return neutral[46];
```
Comment (AshCorr) [inline dotcom-rendering/src/lib/decideContainerOverrides.ts:49]: Can we just remove the case for Branded now?

File: dotcom-rendering/index.d.ts (hunk 4 above)
Comment (alinaboghiu) [reply dotcom-rendering/index.d.ts:327]: this was not being used - sorry to cause extra noise in this PR

File: dotcom-rendering/src/components/Carousel.importable.tsx (hunk 6 above)
Comment (alinaboghiu) [reply dotcom-rendering/src/components/Carousel.importable.tsx]: we have mobbed on this and have refactored - it's much neater now, thanks Olly and Charlotte!

File: dotcom-rendering/src/lib/decideContainerOverrides.ts (hunk 5 above)
Comment (alinaboghiu) [reply dotcom-rendering/src/lib/decideContainerOverrides.ts]: we have agreed to do this as a follow up refactor - I think it's a lot neater so thank you for this suggestion

File: dotcom-rendering/src/lib/decideContainerOverrides.ts (hunk 7 above)
Comment (alinaboghiu) [reply dotcom-rendering/src/lib/decideContainerOverrides.ts:49]: I guess the idea here was that we have made an explicit decision for 'Branded' but not for the other palettes, let's discuss in this afternoon session if you can join at 2? - happy to remove tho it's probably pre-empting too much?

File: dotcom-rendering/src/components/Card/components/LI.tsx [hunk 8]
```diff
@@ -64,10 +65,18 @@ const decideDivider = (
...
-	offsetBottomPaddingOnDivider
-		? verticalDividerWithBottomOffset(paddingSize, containerPalette)
-		: verticalDivider(containerPalette);
+	verticalDividerColour?: string,
+) => {
+	const borderColour =
+		verticalDividerColour ??
+		(containerPalette &&
+			decideContainerOverrides(containerPalette).border.container) ??
+		neutral[86];
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Card/components/LI.tsx:74]: If we’re importing this, should we use the non-deprecated version: `palette.neutral[86]`?

File: dotcom-rendering/src/components/Card/components/UL.tsx [hunk 9]
```diff
@@ -48,11 +49,16 @@ export const UL = ({
 	wrapCards = false,
 	containerPalette,
 }: Props) => {
+	const borderColour =
+		(containerPalette &&
+			decideContainerOverrides(containerPalette).border.container) ??
+		neutral[86];
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Card/components/UL.tsx:55]: Same as in `LI`, should we use the non-deprecated import?

**Review by mxdvl — COMMENTED** [review]
Great changes. Any change you could rewrite the commit history to reduce noise?

**Review by mxdvl — COMMENTED** [review]
––

File: dotcom-rendering/src/components/Card/Card.tsx [hunk 10]
```diff
@@ -83,6 +83,22 @@ export type Props = {
 	showLivePlayable?: boolean;
 };
 
+const getMediaType = (
+	design: Extract<
+		ArticleDesign,
+		ArticleDesign.Gallery | ArticleDesign.Audio | ArticleDesign.Video
+	>,
```
[the remark this answers is not in the document]
Comment (alinaboghiu) [reply dotcom-rendering/src/components/Card/Card.tsx]: will address second point in the next PR

File: dotcom-rendering/src/lib/decideContainerOverrides.ts (hunk 7 above)
Comment (alinaboghiu) [reply dotcom-rendering/src/lib/decideContainerOverrides.ts:49]: we think this will be addressed as part of the refactor mentioned by Max above ^

**Review by mxdvl — APPROVED** [review]
Let’s squash & merge!
